src/main/java/com/hubspot/dropwizard/guicier/GuiceBundle.java

```diff
@@ -202,4 +204,21 @@ private Builder(final Class<U> configClass) {
                                injectorFactory);
     }
   }
+
+  private static String getNextServiceLocatorName() {
+    Class<ServiceLocatorFactoryImpl> factoryClass = ServiceLocatorFactoryImpl.class;
+    try {
+      Field nameCountField = factoryClass.getDeclaredField("name_count");
+      nameCountField.setAccessible(true);
+      int count = (int) nameCountField.get(null);
```

Comment:
This field is also protected by a lock. It seems very unlikely that there'll be a race condition between our code and the `ServiceLocatorFactoryImpl` and if there is, using the lock won't ensure that we get the correct count.